src/components/SourceCodeEditor.tsx

```diff
@@ -32,17 +32,37 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
     : '';
 
   useEffect(() => {
+    let tmpInt: IIntegration = integrationJson;
+    const tmpSettings = settings;
     if (previousJson === integrationJson) return;
-    let tmpInt = integrationJson;
-    tmpInt.metadata = { ...integrationJson.metadata, ...settings };
-    fetchIntegrationSourceCode(tmpInt, settings.namespace).then((newSrc) => {
+
+    if (tmpInt.dsl != null && tmpInt.dsl !== settings.dsl.name) {
+      fetchCapabilities().then((capabilities: ICapabilities) => {
+        capabilities.dsls.forEach((dsl) => {
+          if (dsl.name === tmpInt.dsl) {
+            tmpSettings.dsl = dsl;
+            setSettings(tmpSettings);
+            fetchTheSourceCode(tmpInt, tmpSettings);
+            return;
+          }
+        });
+      });
+    } else {
+      fetchTheSourceCode(integrationJson, settings);
+    }
+  }, [integrationJson]);
+
+  const fetchTheSourceCode = (integration: IIntegration, settings: ISettings) => {
+    integration.metadata = { ...integrationJson.metadata, ...settings };
+    integration.dsl = settings.dsl.name;
```

Comment:
And here.. instead of `integration` below, do a new object like
```
{ ...integration, metadata: {...integrationJson.metadata, ...settings, dsl: settings.dsl.name} }
```
Of course test it first though because I'm just writing this on GitHub 😅